src/main/java/ro/utcn/sd/mdantonio/StackUnderflow/controller/ConsoleController.java

```diff
@@ -0,0 +1,118 @@
+package ro.utcn.sd.mdantonio.StackUnderflow.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import ro.utcn.sd.mdantonio.StackUnderflow.entities.UnderflowUser;
+import ro.utcn.sd.mdantonio.StackUnderflow.exception.*;
+import ro.utcn.sd.mdantonio.StackUnderflow.service.PostManagementService;
+import ro.utcn.sd.mdantonio.StackUnderflow.service.TagManagementService;
+import ro.utcn.sd.mdantonio.StackUnderflow.service.UnderflowUserManagementService;
+
+import javax.persistence.Transient;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+@Component
+@RequiredArgsConstructor
+@Profile("!test")
+public class ConsoleController implements CommandLineRunner {
+    private final Scanner scanner = new Scanner(System.in);
+    private UnderflowUser currentUser = null;
+    private final PostManagementService postManagementService;
+    private final TagManagementService tagManagementService;
+    private final UnderflowUserManagementService underflowUserManagementService;
+
+    @Transient
+    private final List<CommandHandler> commandHandlers = new ArrayList<>();
+
+
+    @Override
+    public void run(String... args) throws Exception {
+
+        commandHandlers.add(new PostManagementCommandHandler(postManagementService));
```

Comment:
You can actually transform the handlers into `@Components` and inject all of them (via Spring). See https://vladmihalcea.com/why-i-like-spring-autowired-for-list-types/